File: Stack/EvaluatePostfix.py
from InfixToPostfix import Stack

def isOperand(c):
    return c not in "+-*/"

def Evaluate(postfix):
    st = Stack()
    st.s = [0] * len(postfix)
    for i in postfix:
        if isOperand(i):
            st.push(int(i))
        else:
            if i=='+':
                c = st.pop() + st.pop()
                st.push(c)
            elif i=='-':
                temp = st.pop()
                c = st.pop() - temp
                st.push(c)
            elif i=='*':
                c = st.pop() * st.pop()
                st.push(c)
            elif i=='/':
                temp = st.pop()
                c = st.pop() / temp
                st.push(int(c))
    return st.stackTop()

print(Evaluate('35*62/+4-'))

# 3*5 + 6/2 - 4
# 15 + 3 -4
# 14

File: Stack/EvaluatePostfix.c
#include "stdio.h"
#include "stdlib.h"
#include "string.h"

struct Node{
    int data;
    struct Node *next;
};

struct Node *top = NULL;

void push(int val){
    struct Node *newNode;
    newNode = (struct Node *)malloc(sizeof(struct Node));
    newNode->data = val;
    newNode->next = top;
    top = newNode;
}

int pop(){
    int x = -1;
    if (top==NULL)
        printf("Stack underflow!\n");
    else{
        x = top->data;
        top = top->next;
    }
    return x;
}

int isOperand(char x){
    if (x=='+'||x=='-'||x=='*'||x=='/')
        return 0;
    else
        return 1;
}

int Evaluate(char *postfix){
    int x,temp;
    for(int i=0;postfix[i]!='\0';i++){
        if (isOperand(postfix[i])){
            push(postfix[i]-'0');
        }
        else {
            if (postfix[i]=='+'){
                x = pop() + pop();}
            else if (postfix[i]=='-'){
                temp = pop();
                x = pop() - temp;
            } else if(postfix[i]=='*'){
                x = pop() * pop();
            } else if(postfix[i]=='/'){
                temp = pop();
                x = pop() / temp;
            }
            push(x);
        }
    }
    return top->data;
}

int main(){
    char *postfix = "234*+82/-";
    printf("%d", Evaluate(postfix));
    return 0;
}